client/views/teams/contextualBar/ChannelDesertionTable.js

```diff
@@ -17,35 +17,75 @@ const ChannelDesertionTable = ({
 	lastOwnerWarning,
 }) => {
 	const t = useTranslation();
-
+	const [sort, setSort] = useState(['name', 'asc']);
 	const selectedRoomsLength = Object.values(selectedRooms).filter(Boolean).length;
 
 	const checked = eligibleRoomsLength === selectedRoomsLength;
 	const indeterminate = eligibleRoomsLength > selectedRoomsLength && selectedRoomsLength > 0;
 
 	const formatDate = useFormatDateAndTime();
 
+	const onHeaderClick = useCallback(
+		(id) => {
+			const [sortBy, sortDirection] = sort;
+			if (sortBy === id) {
+				setSort([id, sortDirection === 'asc' ? 'desc' : 'asc']);
+				return;
+			}
+			setSort([id, 'asc']);
+		},
+		[sort],
+	);
+
+	const getSortedChannels = useCallback(() => {
```

Comment:
This could be moved to outside the function in order to avoid using `useCallback`. Then you could call `getSortedChannels(rooms, sort)`.